MaxText/max_utils.py

```diff
@@ -167,10 +167,16 @@ def init_train_state(model, tx, config, key):
 
   Args: model, tx, config, key
   """
-  input_shape = (
-      len(jax.devices()) * config.per_device_batch_size,
-      config.max_target_length
-  )
+  if config.per_device_batch_size < 1:
```

Comment:
since you keep doing this calculation, should it be shared with the above. Something like "calculate_raw_input_batch_size()" as a function.

Reply:
Done.

Reply:
Done.